driver/ni/task.cpp

```diff
@@ -183,179 +193,163 @@ void ni::ReaderTask::exec(task::Command &cmd) {
+        .base_interval = 1 * SECOND,
+        .max_retries = 20,
+        .scale = 1.2,
     };
 
-    LOG(INFO) << "[NI Task] configuring task " << task.name;
+    auto parser = config::Parser(task.config);
+    auto data_saving = parser.optional<bool>("data_saving", true);
 
     TaskHandle task_handle;
-    // create a daq reader to provide to cmd read pipe as sink
     ni::NiDAQmxInterface::CreateTask("", &task_handle);
-    //print taskhandle
-    LOG(INFO) << "Task handle: " << task_handle;
 
-    auto daq_writer = std::make_unique<ni::DigitalWriteSink>(task_handle, ctx, task);
-    if (!daq_writer->ok()) {
-        LOG(ERROR) << "[NI Writer] failed to construct reader for" << task.name;
-        return nullptr;
-    }
+    LOG(INFO) << "[ni.writer] configuring task " << task.name;
+    auto daq_writer = std::make_shared<ni::DigitalWriteSink>(task_handle, ctx, task);
 
     // construct writer config
-    std::vector <synnax::ChannelKey> cmd_keys = daq_writer->getCmdChannelKeys();
-    std::vector <synnax::ChannelKey> state_keys = daq_writer->getStateChannelKeys();
+    std::vector<synnax::ChannelKey> cmd_keys = daq_writer->get_cmd_channel_keys();
+    std::vector<synnax::ChannelKey> state_keys = daq_writer->get_state_channel_keys();
 
     // create a writer config to write state channels
     auto writer_config = synnax::WriterConfig{
-            .channels = state_keys,
-            .start = synnax::TimeStamp::now(),
-            .enable_auto_commit = true,
+        .channels = state_keys,
+        .start = synnax::TimeStamp::now(),
+        .mode = data_saving 
+                    ? synnax::WriterMode::PersistStream 
+                    : synnax::WriterMode::StreamOnly,
+        .enable_auto_commit = true,
     };
 
     // create a streamer config to stream incoming cmds
     auto streamer_config = synnax::StreamerConfig{
-            .channels = cmd_keys,
-            .start = synnax::TimeStamp::now(),
+        .channels = cmd_keys,
+        .start = synnax::TimeStamp::now(),
     };
 
+
+    auto state_writer = daq_writer->writer_state_source;
+
+    daq_writer->cycle();
+    LOG(INFO) << "[ni.writer] constructed writer for " << task.name;
+    auto p = std::make_unique<ni::WriterTask>(ctx,
+                                              task,
+                                              daq_writer,
+                                              daq_writer,
+                                              state_writer,
+                                              writer_config,
+                                              streamer_config,
+                                              breaker_config);
+
+    if (!daq_writer->ok()) {
+        LOG(ERROR) << "[ni.writer] failed to construct writer for " << task.name;
+        return p;
+    }
+
+    // sleep for 10ms: this is here to temporarily fix a race condition in the console after hitting configure
+    std::this_thread::sleep_for(std::chrono::milliseconds(10));
```

Comment:
Remove this. This was fixed in the console, and was, in fact, not a race condition :)